### PR TITLE
Update py-redis

### PR DESCRIPTION
**Pull request by koning**
Required for py-celery and py-merlin-1.7.5 update.

### CODE REVIEW

**Comment by koning** [comment]
Not sure what to do about the failure on this one.

**Review by scheibelp — APPROVED** [review]

**Comment by koning** [comment]
Conflict resolved.

File: var/spack/repos/builtin/packages/py-redis/package.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class PyRedis(PythonPackage):
     homepage = "https://pypi.org/project/redis/"
     url      = "https://pypi.io/packages/source/r/redis/redis-3.3.8.tar.gz"
 
+    version('3.5.3', sha256='0e7e0cfca8660dea8b7d5cd8c4f6c5e29e11f31158c0b0ae91a397f00e5a05a2')
     version('3.5.0', sha256='7378105cd8ea20c4edc49f028581e830c01ad5f00be851def0f4bc616a83cd89')
     version('3.3.8', sha256='98a22fb750c9b9bb46e75e945dc3f61d0ab30d06117cbb21ff9cd1d315fedd3b')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-redis/package.py:18]: Can you update the dependencies? It looks like Python is now 2.7+ and 3.5+. Hiredis is actually optional, and requires 0.1.3+.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-redis/package.py (hunk 1 above)
Comment (koning) [reply var/spack/repos/builtin/packages/py-redis/package.py:18]: Sure, sorry I didn't catch that.

File: var/spack/repos/builtin/packages/py-redis/package.py [hunk 2]
```diff
@@ -12,9 +12,13 @@ class PyRedis(PythonPackage):
...
 
-    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
-    depends_on('py-setuptools',       type='build')
-    depends_on('py-hiredis',          type=('build', 'run'))
+    variant("hiredis", "Support for hiredis which speeds up parsing of multi bulk replies")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-redis/package.py]:
```suggestion
    variant("hiredis", default=False, description="Support for hiredis which speeds up parsing of multi bulk replies") 
```
I'll let you decide what the default should be.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**